quickwit-doc-mapping/src/default_doc_mapper/field_mapping_type.rs

```diff
@@ -0,0 +1,77 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use tantivy::schema::{BytesOptions, Cardinality, IntOptions, TextOptions};
+
+use super::FieldMappingEntry;
+
+/// A `FieldMappingType` defines the type and indexing options
+/// of a mapping field.
+#[derive(Clone, Debug)]
+pub enum FieldMappingType {
+    /// String mapping type configuration.
+    Text(TextOptions),
+    /// Signed 64-bits integers mapping type configuration.
+    I64(IntOptions),
+    /// 64-bits floats mapping type configuration.
+    F64(IntOptions),
+    /// Signed 64-bits Date 64 mapping type configuration,
+    Date(IntOptions),
+    /// Bytes mapping type configuration.
+    Bytes(BytesOptions),
+    /// Object mapping type configuration.
+    Object(Vec<FieldMappingEntry>),
+}
+
+impl FieldMappingType {
+    pub fn type_with_cardinality(&self) -> String {
+        let is_array = match &self {
+            FieldMappingType::I64(options)
+            | FieldMappingType::Date(options)
+            | FieldMappingType::F64(options) => {
+                options
+                    .get_fastfield_cardinality()
+                    .unwrap_or(Cardinality::SingleValue)
+                    == Cardinality::MultiValues
+            }
+            FieldMappingType::Text(_)
+            | FieldMappingType::Bytes(_)
+            | FieldMappingType::Object(_) => false,
+        };
+        if is_array {
+            format!("array<{}>", self.field_type())
+        } else {
+            self.field_type().to_string()
+        }
+    }
+
+    fn field_type(&self) -> &str {
```

Comment:
The method name suggest it returns a FieldType.

Can we make it `field_type_str` or something?